langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiImageModel.java

```diff
@@ -0,0 +1,149 @@
+ */
+public class OpenAiImageModel implements ImageModel {
+
+    private final String model;
+    private final Integer n;
+    private final String size;
+    private final String quality;
+    private final String style;
+    private final String user;
+    private final String responseFormat;
+
+    private final OpenAiClient client;
+
+    private final Integer maxRetries;
+
+    /**
+     * Instantiates OpenAI DALL·E image processing model.
+     * Find the parameters description <a href="https://platform.openai.com/docs/api-reference/images/create">here</a>.
+     *
+     * @param model          dall-e-3 is default one
+     * @param persistTo      specifies the local path where the generated image will be downloaded to (in case provided).
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     * @param withPersisting generated response will be persisted under <code>java.io.tmpdir</code>.
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     */
+    @Builder
+    @SuppressWarnings("rawtypes")
+    public OpenAiImageModel(
+        @NonNull String apiKey,
+        String model,
+        Integer n,
+        String size,
+        String quality,
+        String style,
+        String user,
+        String responseFormat,
+        Duration timeout,
+        Integer maxRetries,
+        Proxy proxy,
+        Boolean logRequests,
+        Boolean logResponses,
+        Boolean withPersisting,
+        Path persistTo
+    ) {
+        timeout = getOrDefault(timeout, ofSeconds(60));
+
+        OpenAiClient.Builder cBuilder = OpenAiClient
+            .builder()
+            .openAiApiKey(apiKey)
+            .callTimeout(timeout)
+            .connectTimeout(timeout)
+            .readTimeout(timeout)
+            .writeTimeout(timeout)
+            .proxy(proxy)
+            .logRequests(getOrDefault(logRequests, false))
+            .logResponses(getOrDefault(logResponses, false))
+            .persistTo(persistTo);
+
+        if (withPersisting != null && withPersisting) {
+            cBuilder.withPersisting();
+        }
+
+        this.client = cBuilder.build();
+
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.model = model;
+        this.n = n;
+        this.size = size;
+        this.quality = quality;
+        this.style = style;
+        this.user = user;
+        this.responseFormat = responseFormat;
+    }
+
+    public static class OpenAiImageModelBuilder {
+
+        public OpenAiImageModelBuilder withPersisting() {
+            withPersisting = true;
+            return this;
+        }
+
+        public OpenAiImageModelBuilder logRequests() {
+            logRequests = true;
```

Comment:
This is a really nice change but I would propose to make it in a separate PR and update all model classes at once to keep the consistency

Reply:
done